doc/manual/src/language/index.md

```diff
@@ -4,9 +4,7 @@ The Nix language is
 
 - *domain-specific*
 
-  It only exists for the Nix package manager:
-  to describe packages and configurations as well as their variants and compositions.
-  It is not intended for general purpose use.
+  Its purpose is to conveniently create and compose precise descriptions of how contents of existing files are used to derive new files.
```

Comment:
```suggestion
  Its purpose is to conveniently create and compose precise descriptions of how contents of existing files are used to derive new files.
  It comes with [builitin functions](@docroot@/language/builtins.md) to integrate with the heart of Nix. 
```

"Purpose" is somewhat an opinion too. But what is a fact is that the Nix language, which is otherwise domain-agnostic, is connected to the store layer, which is the heart of Nix, via the builtins.